src/test/java/com/puppycrawl/tools/checkstyle/utils/XpathUtilTest.java

```diff
@@ -120,7 +123,9 @@ public void testPrintXpathNotComment() throws Exception {
     @Test
     public void testPrintXpathComment() throws Exception {
         final String fileContent = "class Test { /* comment */ }";
-        final File file = File.createTempFile("junit", null, tempFolder);
+        final String uniqueFileName = "junit_" + UUID.randomUUID() + ".java";
+        final File file =
+                Files.createFile(tempFolder.toPath().resolve(uniqueFileName)).toFile();
         Files.write(file.toPath(), fileContent.getBytes(StandardCharsets.UTF_8));
```

Comment:
Either with the previous or without, something needs to be done with all these file to path back to file and back to path again. Please clear it all up.

Reply:
@rnveach, if i understood you right, the below code can help us resolve the back and forth between File and Path object:
```
        final File file = new File(tempFolder, uniqueFileName);
        Files.write(file.toPath(), fileContent.getBytes(StandardCharsets.UTF_8));
```

Reply:
It is compact, l am ok with this, let's apply it

Reply:
> It is compact, l am ok with this, let's apply it

Sure, waiting for @rnveach feedback, and will implement the changes.

Reply:
It will be quicker to just do it, as it reasonable, elegant and compact.

Reply:
> It will be quicker to just do it, as it reasonable, elegant and compact.

But, there is one nuance here. Using the constructor doesn't create an actual file. And, some test cases require the explicit creation of file. `createFile `used to create actual file.

Reply:
> It will be quicker to just do it, as it reasonable, elegant and compact.

Done. Except in some files, I have kept original `createFile` because the code of the tests expect acutal file. And, the constructor apporach just creates an object rather an actual file.

Reply:
You should do what is needed to resolve this. If you think you are done, or can't see how to resolve it then let me know.

Reply:
@rnveach sorry for the ping. are the changes good enough?